File: HighPerformaceTest/HighPerformaceWindowsApp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace HighPerformaceWindowsApp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            StringBuilderPerformace();
        }

        public String StringBuilderPerformace()
        {
            StringBuilder sb = new StringBuilder();
            String str = "";
            int start = DateTime.Now.Millisecond;
            for (int i = 0; i < 100000; i++)
            {
                sb.Append(i);
                sb.Append(", ");
            }
            str = sb.ToString();
            int end = DateTime.Now.Millisecond;
            //Console.WriteLine(str);
            Console.WriteLine("Took: " + (end-start) +" ms.");
            
            return "";
        }
    }
}

File: HighPerformaceTest/VBHighPerfomaceWindowsApplication/Form1.vb
Imports System
Imports System.Text
Public Class Form1
    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        StringBuilderPerform()
        ConcateStringByPlus()
    End Sub

    Sub StringBuilderPerform()
        Console.WriteLine("start")
        Dim sb As New StringBuilder()
        Dim startTime As DateTime = DateTime.Now
        For i As Integer = 1 To 10000
            sb.Append(i)
            sb.Append(", ")
        Next
        Dim endTime As DateTime = DateTime.Now
        Console.WriteLine("String Builder Took: " & (endTime - startTime).ToString)
    End Sub
    Sub ConcateStringByPlus()
        Console.WriteLine("start")
        Dim sb As New StringBuilder()
        Dim str As String = ""
        Dim startTime As DateTime = DateTime.Now
        For i As Integer = 1 To 10000
            str += i.ToString
        Next
        Dim endTime As DateTime = DateTime.Now
        Console.WriteLine("+= Took: " & (endTime - startTime).ToString)
    End Sub
End Class
